misc/lldb_disasm.py

```diff
@@ -0,0 +1,196 @@
+            print("VARIABLE %0#x" % op, end = "", file=result)
+        elif op_type == TS_CALLDATA:
+            print("(struct rb_call_data *)%0#x" % op, end = "", file=result)
+        elif op_type == TS_CDHASH:
+            print("CDHASH (VALUE)%0#x" % op, end = "", file=result)
+        elif op_type == TS_IC:
+            print("(struct iseq_inline_cache_entry *)%0#x" % op, end = "", file=result)
+        elif op_type == TS_IVC:
+            print("(struct iseq_inline_iv_cache_entry *)%0#x" % op, end = "", file=result)
+        elif op_type == TS_ID:
+            print("ID: %0#x" % op, end="", file=result)
+        elif op_type == TS_ISE:
+            print("(iseq_inline_storage_entry *)%0#x" % op, end = "", file=result)
+        elif op_type == TS_ISEQ:
+            print("(rb_iseq_t *)%0#x" % op, end = "", file=result)
+        elif op_type == TS_OFFSET:
+            print("%d" % op, end="", file=result)
+        elif op_type == TS_VALUE:
+            print("(VALUE)%0#x" % op, end = "", file=result)
+        elif op_type == TS_LINDEX:
+            print("%d" % op, end="", file=result)
+        elif op_type == TS_FUNCPTR:
+            print("(rb_insn_func_t) %0#x" % op, end = "", file=result)
+        elif op_type == TS_NUM: 
+            print("%d" % op, end="", file=result)
+        elif op_type == TS_BUILTIN: 
+            print("(rb_builtin_function *)%0#x" % op, end="", file=result)
+        else:
+            print("UNREACHABLE?" % op, end="", file=result)
+
+        idx += 1
+
+    print(" )", file=result)
+    return length
+
+def insn_len(target, offset):
+    tChar = target.FindFirstType("char")
+    size_of_char = tChar.GetByteSize();
+
+    addr_of_table = target.FindSymbols("insn_len.t")[0].GetSymbol().GetStartAddress().GetLoadAddress(target)
+
+    addr_in_table = addr_of_table + (offset * size_of_char)
+    addr = lldb.SBAddress(addr_in_table, target)
+
+    return target.CreateValueFromAddress("y", addr, tChar).GetValueAsUnsigned()
+
+def insn_op_types(target, process, result, insn):
+    tUShort = target.FindFirstType("unsigned short")
+    tChar   = target.FindFirstType("char")
+
+    size_of_short = tUShort.GetByteSize();
+    size_of_char =  tChar.GetByteSize();
+
+    addr_of_table = target.FindSymbols("insn_op_types.y")[0].GetSymbol().GetStartAddress().GetLoadAddress(target)
+    addr_in_table = addr_of_table + (insn * size_of_short)
+    addr = lldb.SBAddress(addr_in_table, target)
+    offset = target.CreateValueFromAddress("y", addr, tUShort).GetValueAsUnsigned()
+
+    addr_of_table = target.FindSymbols("insn_op_types.x")[0].GetSymbol().GetStartAddress().GetLoadAddress(target)
+    addr_in_name_table = addr_of_table + (offset * size_of_char)
+
+    error = lldb.SBError()
+    return process.ReadCStringFromMemory(addr_in_name_table, 256, error)
+
+def insn_name_table_offset(target, offset):
+    tUShort = target.FindFirstType("unsigned short")
+    size_of_short = tUShort.GetByteSize();
```

Comment:
`;` could be deleted

Reply:
argh!  Thanks for pointing these out. I guess I was flipping between C and Python too much 🤦🏻‍♂️